### ISSUE TITLE
Trouble sourcing plugin file via zinit

### ISSUE THREAD

**Issue by olets** [romkatv/gitstatus#125]
I'm working on a gitstatus-enabled plugin, the essential logic of which follows `gitstatus.prompt.zsh`:

```
myfunc() {
  gitstatus_query 'MY' || return 1  # error
}
gitstatus_stop 'MY' && gitstatus_start -s -1 -u -1 -c -1 -d -1 'MY'
autoload -Uz add-zsh-hook
add-zsh-hook precmd myfunc
```

I know in the gitstatus README you recommend `source`ing the gitstatus plugin file, but to try to squeeze a little extra performance and to simplify the installation experience for when it goes public I've been doing

```
zinit light romkatv/gitstatus
zinit load path/to/my/prompt
```

That's worked for the last couple months. Today I updated (with `zinit update`) from cc7fc58 to c673d32 [just now updated to ae98815 and this error persists], opened a new session, and got the errors

```
command not found: gitstatus_stop
command not found: gitstatus_query
```

(I haven't looked into what changed between the two commits.)

If I drop the `gitstatus_stop 'MY' &&` I get

```
command not found: gitstatus_start
command not found: gitstatus_query
```

I tried

```
zinit ice pick"gitstatus.plugin.zsh"
zinit light romkatv/gitstatus
zinit load path/to/my/prompt
```

and

```
zinit ice src"gitstatus.plugin.zsh"
zinit light romkatv/gitstatus
zinit load path/to/my/prompt
```

without success.

I was able to work around this with

```
source ~/.zinit/plugins/romkatv---gitstatus/gitstatus.plugin.zsh
zinit load path/to/my/prompt
```

The fix makes me wonder if it's a zinit bug (I also updated zinit), but I'm reporting here first because I'm not getting errors related to any other plugins I'm managing with zinit. Also open to the possibility that this never should have worked, but I'm not sure why that would be.

**Comment by romkatv** [comment]
Thanks for the report. I've sent a PR to zinit to fix this: https://github.com/zdharma/zinit/pull/336. In the meantime you'll have to use `source`.

**Comment by romkatv** [comment]
By the way, is your code that uses gitstatus public? Could you share a link?

**Comment by romkatv** [comment]
> Thanks for the report. I've sent a PR to zinit to fix this: [zdharma/zinit#336](https://github.com/zdharma/zinit/pull/336). In the meantime you'll have to use `source`.

FYI: This has been merged.

> By the way, is your code that uses gitstatus public? Could you share a link?

This question is still relevant.

**Comment by olets** [comment]
Thanks!

Not public yet but soon, I'll let you know.

> accepts an optional argument when being sourced. This argument gets added to the names of all functions and global variables that get defined during the script's evaluation. This is meant to make it easier to embed gitstatus within other plugins (https://github.com/zdharma/zinit/pull/336#issuecomment-627786311)

Hadn't noticed that, smart I'll check it out
